include/swift/Parse/Parser.h

```diff
@@ -1515,10 +1514,6 @@ class Parser {
     /// all the arguments, not just those that have default arguments.
     unsigned claimNextIndex() { return NextIndex++; }
```

Comment:
Seems like `claimNextIndex` is unused?

Reply:
Would also be nice to see if we could remove `HasDefaultArgument`, seems like the diagnostic logic there could either be plumbed down or check the ParameterList itself

Reply:
Yeah, `DefaultArgumentInfo` does mostly nothing anymore. I guess `HasDefaultArgument` diagnostic can be sinked into `TyoeCheckDeclPrimary`. I will try it in followups, those diagnostics are needed in ASTGen mode too.